h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -421,7 +421,7 @@ public final int close() {
           // For small-packet write, send via UDP.  Since nothing is sent until
           // now, this close() call trivially orders - since the reader will not
           // even start (much less close()) until this packet is sent.
-          if( _bb.position() < MTU) return udpSend();
+          if( _bb.limit() < MTU) return udpSend();
```

Comment:
@jakubhava please see the implementation of `udpSend`:
```
  private int udpSend() throws IOException {
    assert _chan == null;
    TimeLine.record_send(this,false);
    _size = _bb.position();
    assert _size < AutoBuffer.BBP_SML._size;
    _bb.flip();                 // Flip for sending
```

the assumption here is that `size == position`, also `bb.flip` does set `limit = position` - IMHO this is therefore not correct way to fix this

Reply:
We can go to `!__bb.isDirect() && _bb.position() < MTU` I was thinking about it and I thin it actually descrives the invariant more explicitly, what do you think?

Reply:
that might be better, but the invariant that `_bb.position == size` is still used later 

and we really shouldn't have a Direct BB that is small

there must be a bug elsewhere,

didn't you say that the size of the data is 65k? it is suspiciously close to

static BBPool BBP_BIG = new BBPool(64*1024);